src/main/kotlin/com/terraformation/backend/accelerator/db/ReportStore.kt

```diff
@@ -1462,35 +1465,94 @@ class ReportStore(
           )
           .convertFrom { it.toInt() }
 
-  private val plotHasCompletedPermanentObservations =
+  private fun plotHasCompletedObservations(
+      monitoringPlotIdField: Field<MonitoringPlotId?>,
+      isPermanent: Boolean,
+  ): Condition =
       DSL.exists(
           DSL.selectOne()
-              .from(OBSERVATION_PLOTS)
-              .where(OBSERVATION_PLOTS.MONITORING_PLOT_ID.eq(PLOT_T0_DENSITIES.MONITORING_PLOT_ID))
-              .and(OBSERVATION_PLOTS.OBSERVATION_ID.`in`(observationsInReportPeriod))
-              .and(OBSERVATION_PLOTS.IS_PERMANENT.eq(true))
-              .and(OBSERVATION_PLOTS.COMPLETED_TIME.isNotNull)
+              .from(
+                  DSL.select(OBSERVATION_PLOTS.IS_PERMANENT)
+                      .from(OBSERVATION_PLOTS)
+                      .where(
+                          OBSERVATION_PLOTS.MONITORING_PLOT_ID.eq(monitoringPlotIdField)
+                              .and(OBSERVATION_PLOTS.COMPLETED_TIME.isNotNull)
+                      )
+                      .and(OBSERVATION_PLOTS.OBSERVATION_ID.`in`(observationsInReportPeriod))
+                      .orderBy(
+                          OBSERVATION_PLOTS.COMPLETED_TIME.desc(),
+                          OBSERVATION_PLOTS.OBSERVATION_ID.desc(),
+                      )
+                      .limit(1)
+                      .asTable("most_recent")
+              )
+              .where(DSL.field("most_recent.IS_PERMANENT", Boolean::class.java).eq(isPermanent))
```

Comment:
I saw a similar query in an earlier PR and wondered if it could be made simpler with a window function. Couldn't get that to come out any simpler than this version, but it just occurred to me that maybe this can use something along the lines of
```
DSL.select(IS_PERMANENT.eq(isPermanent))
```
instead of the where/exists to at least get rid of one layer of query nesting. Maybe wrapped in a coalesce to deal with the case of no matching rows.

The query as it exists works fine and isn't hard to understand, but for some reason every time I see this, I get the urge to try to simplify it. Feel free to ignore this comment if you don't think it's worth fussing with.